tests/lang_callable/macros/tmacros_issues.nim

```diff
@@ -519,3 +519,16 @@ block double_sem_for_procs:
     result = 10.0
 
   discard exp(5.0)
+
+block orginal_parameter_types:
+  # inspecting a parameter type in the macro's header yields the
+  # original type, not ``NimNode``
+  macro m(x: int, y: typeof(x)) =
```

Comment:
```suggestion
  template assertTypeOfInt(s): typedesc =
    doAssert typeof(s) is int
    typeof(s)
  # inspecting a parameter type in the macro's header yields the
  # original type, not ``NimNode``
  macro m(x: int, y: assertTypeOfInt(x)) =
```

It's a very narrow case, but this might help detect issues, if there were a cascade of bugs resulting in: `typeof(x)` is evaluated to `NimNode` and things get passed through for `y` regardless.

Reply:
Good call; I updated it.